lib/Command/UpdateGroup.php

```diff
@@ -196,25 +197,34 @@ private function updateGroupMapping($groupName, $userList) {
 	}
 
 	/**
-	 * Make sure $groupNames doesn't contain duplicated values. This function could behave
-	 * unexpectedly otherwise.
+	 * Split the list of group names into those which are in the DB and those which aren't
+	 * As an example, checkGroupMappingExists(['g1', 'g2', 'g3', 'missing', 'another']) could return:
+	 * ['inDB' => ['g1', 'g2', 'g3'], 'notinDB' => ['missing', 'another']]
 	 *
-	 * Take advantage of the owncloud_name column in the DB has a unique constraint.
-	 *
-	 * @return true if the count($groupNames) matches the number of
+	 * @param array<String> $groupNames list of group names
+	 * @return array returns an array containing 2 arrays: 'inDB' for those group names that are in
+	 * the DB and 'notinDB' for those that aren't
 	 */
 	private function checkGroupMappingExists($groupNames) {
+		$groups = array('inDB' => array(), 'notinDB' => array());
 		$query = $this->connection->getQueryBuilder();
-		$query->select($query->createFunction('count(owncloud_name) as ngroups'))
+		$query->select('owncloud_name')
 			->from('ldap_group_mapping')
 			->where($query->expr()->in('owncloud_name', $query->createParameter('groups')))
 			->setParameter('groups', $groupNames, IQueryBuilder::PARAM_STR_ARRAY);
 		$result = $query->execute();
-		$row = $result->fetch();
 
-		$countValue = intval($row['ngroups']);
+		// fill with those which are inside the DB
+		while (($row = $result->fetch()) !== false) {
+			if (in_array($row['owncloud_name'], $groupNames, true)) {
+				$groups['inDB'][] = $row['owncloud_name'];
```

Comment:
Ah never mind, this is only based on the input list, so unless some admin passes 1000 groups to the CLI it should be fine.